core/src/err.rs

```diff
@@ -23,22 +23,22 @@ pub enum Error {
     MissingFieldBsonType(String),
     #[error("Invalid connection string. Parse error: {0}")]
     MongoParseConnectionString(mongodb::error::Error),
-    #[error(transparent)]
-    Mongo(#[from] mongodb::error::Error),
+    #[error("Trying to {0} gave: {1}")]
+    Mongo(String, mongodb::error::Error),
     #[error("No database provided for query")]
     NoDatabase,
     #[error("Unknown column '{0}' in result set schema")]
     UnknownColumn(String),
-    #[error(transparent)]
-    ValueAccess(bson::document::ValueAccessError),
+    #[error("Trying to access collection '{0}' gave: {1}")]
```

Comment:
Umh, the error happens when getting the value for a specific field.
The collection name might not make any sense to the end user here. It is easily empty, for example with `select 1 from mytable` or if there is an alias, this is the collection name we will see.
What about "Error retrieving data for field {0} : {1}", with 0 being the column index?